### PR TITLE
feat(WEB-5967): Upgrade dependencies

### PR DESCRIPTION
**Pull request by georgegillams**
BREAKING CHANGE: Upgrade dependencies

For most, this will probably be a seamless upgrade. But to avoid nasty surprises, I think it's best to class it as breaking just in case.

[WEB-5967](https://typeform.atlassian.net/browse/WEB-5967)

 - eslint-config-prettier
   - All configs are now merged into one
   - https://github.com/prettier/eslint-config-prettier/blob/main/CHANGELOG.md#version-800-2021-02-21
 - eslint-config-standard
   - `eslint-config-node` has been replaced with `eslint-config-n`, so some rule names will have changed
   - https://github.com/prettier/eslint-config-prettier/blob/main/CHANGELOG.md#version-800-2021-02-21
 - eslint-plugin-flowtype
   - Drops node 10 and ESLint v7 support
   - https://github.com/gajus/eslint-plugin-flowtype/releases/tag/v7.0.0
   - https://github.com/gajus/eslint-plugin-flowtype/releases/tag/v8.0.0
 - eslint-plugin-jest
   - Rules renamed https://github.com/jest-community/eslint-plugin-jest/releases/tag/v25.0.0
   - Dropped support for @typescript-eslint/eslint-plugin@4 https://github.com/jest-community/eslint-plugin-jest/releases/tag/v26.0.0
   - Dropped support for node 12, 17. Renamed some rules https://github.com/jest-community/eslint-plugin-jest/releases/tag/v27.0.0
 - eslint-plugin-promise
   - Dropped eslint 7 support https://github.com/eslint-community/eslint-plugin-promise/blob/main/CHANGELOG.md#500
   - Dropped node 10 support https://github.com/eslint-community/eslint-plugin-promise/blob/main/CHANGELOG.md#600

Short summary
---

## Things to include:

- [x] Description of what's changed and why it will help
- [x] Updated test pass case
- [x] Added test fail case

### CODE REVIEW

**Comment by matthizou** [comment]
Nice!
I am a bit nervous, though, to push so many dependency upgrades in one go. We should first try it with a few big projects, see where/if it breaks. Get some confidence. 
Can you release an alpha/beta version of this and test it in Hall of Forms?

If it breaks, we could follow a more gradual approach, first upgrading all the dependencies that just do minor version upgrades.

**Comment by georgegillams** [comment]
> Nice one 👏 ! I am a bit nervous, though, to push so many dependency upgrades in one go. We should first try it with a few big projects, see where/if it breaks. Get some confidence. Can you release an alpha/beta version of this and test it in Hall of Forms?
> 
> If it breaks, we could follow a more gradual approach, first upgrading all the dependencies that just do minor version upgrades.

Thanks for reviewing @matthizou 

I agree it's worth splitting the changes up as it might help other teams adopting the changes.
 - https://github.com/Typeform/eslint-config-typeform/pull/65 only updates the tests in preparation for the new dependencies
 - https://github.com/Typeform/eslint-config-typeform/pull/66 contains minor dependency upgrades and devDependency upgrades
 - Once those are both released, I'll rebase this PR, and we can release this as breaking

I've tested https://github.com/Typeform/eslint-config-typeform/pull/66 and this PR with beta releases (`@typeform/eslint-config@5.2.1-beta.0` and `@typeform/eslint-config@6.0.0-beta.0`). In Hall Of Forms neither causes significant issues — there's one new eslint error (which looks valid to me) and no changes to how prettier runs.

I already tested the changes within the web team ownerships and there were no issues there either 👍

File: package.json [hunk 1]
```diff
@@ -29,32 +29,32 @@
   },
   "prettier": "./prettier",
   "dependencies": {
-    "@typescript-eslint/eslint-plugin": "^5.44.0",
-    "@typescript-eslint/parser": "^5.44.0",
+    "@typescript-eslint/eslint-plugin": "^5.43.0",
+    "@typescript-eslint/parser": "^5.43.0",
```
Comment (matthizou) [inline package.json:33]: Any reason for also including `v5.43`?

File: package.json [hunk 2]
```diff
@@ -29,32 +29,32 @@
...
-    "eslint-plugin-promise": "^4.2.1",
-    "eslint-plugin-react": "^7.31.11",
+    "eslint-plugin-promise": "^6.1.1",
+    "eslint-plugin-react": "^7.31.10",
     "eslint-plugin-react-hooks": "^4.6.0"
   },
   "peerDependencies": {
     "@babel/core": "7.x",
-    "eslint": ">=7.27.x",
+    "eslint": ">=7.32.x",
```
Comment (matthizou) [inline package.json:51]: This is a breaking change for the consumers, needs to be documented in the release

File: package.json (hunk 1 above)
Comment (georgegillams) [reply package.json]: Just a bad rebase. Have now fixed 👍

File: package.json (hunk 2 above)
Comment (georgegillams) [reply package.json]: We don't actually need to change this, so eaiser to leave it as it 👍

**Review by matthizou — APPROVED** [review]

**Pull request merged**